### DISCUSSION TITLE
Import spk error

### DISCUSSION THREAD

**Discussion by ariel234156** [nextpyp/.github#123] [General]
Hi, 
I tried to upload particle locations (spk file) to the preproccess step, and I got this error:

`<html>
<body>

Traceback (most recent call last):
--
155 | File "/opt/pyp/bin/run/pyp", line 3440, in <module>
156 | tomo_swarm(args.path, args.file, args.debug, args.keep, args.skip)
157 | File "/opt/pyp/src/pyp/utils/timer.py", line 78, in wrapper_timer
158 | return func(*args, **kwargs)
159 | File "/opt/pyp/bin/run/pyp", line 1663, in tomo_swarm
160 | virion_coordinates, spike_coordinates = detect_tomo.detect_and_extract_particles(
161 | File "/opt/pyp/src/pyp/detect/tomo/core.py", line 1290, in detect_and_extract_particles
162 | extract_spk_direct(
163 | File "/opt/pyp/src/pyp/detect/tomo/core.py", line 1355, in extract_spk_direct
164 | spikes = imod.coordinates_from_mod_file("%s.spk" % name)
165 | File "/opt/pyp/src/pyp/inout/utils/pyp_edit_box_files.py", line 26, in coordinates_from_mod_file
166 | subprocess.getoutput(
167 | IndexError: list index out of range

</body>
</html>`

The coordinates were compatible with the output tomogram (with eight binning), and the file contained three columns [x,y,z] coordinates.
I would be happy to help.
Thank you.

**Comment by abartesaghi** [comment]
If you are trying to import particle coordinates using `.spk` files, those must be IMOD model files, not simple text files with coordinates. You have two options, either rename the `.txt` files to have a `.box` extension, or preferably, convert your `.txt` files to IMOD models using IMOD's `point2model` program. Having the models will allow you to open each tomogram and its corresponding `.spk` file in IMOD, and confirm that your particles are in the right locations before importing them into nextpyp.

**Comment by ariel234156** [comment]
Thank you, it's working now.
